File: windows/src/GoalMaker.Core/Updates/ManifestCheck.cs
namespace GoalMaker.Core.Updates;

/// <summary>The outcome of checking a release manifest: signature first, then content.</summary>
public abstract record ManifestCheck
{
    private ManifestCheck()
    {
    }

    public sealed record Valid(ReleaseManifest Manifest) : ManifestCheck;

    public sealed record BadSignature : ManifestCheck;

    public sealed record BadManifest(string Reason) : ManifestCheck;
}

File: android/app/src/main/kotlin/com/goalmaker/app/domain/update/ManifestCheck.kt
package com.goalmaker.app.domain.update

/** The outcome of checking a release manifest: signature first, then content. */
sealed interface ManifestCheck {
    data class Valid(val manifest: ReleaseManifest) : ManifestCheck

    data object BadSignature : ManifestCheck

    data class BadManifest(val reason: String) : ManifestCheck
}
